tests/python_functional/src/helpers/snmptrapd/snmptrapd.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import os
+
+import pytest
+from pathlib2 import Path
+from psutil import TimeoutExpired
+
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.driver_io.file.file_io import FileIO
+from src.executors.process_executor import ProcessExecutor
+from src.syslog_ng_config.statements.destinations.destination_reader import DestinationReader
+
+
+class SNMPtrapd(object):
+    def __init__(self, port):
+        self.snmptrapd_proc = None
+        self.port = port
+
+        self.snmptrapd_log = Path(tc_parameters.WORKING_DIR, "snmptrapd_log")
+        self.snmptrapd_stdout_path = Path(tc_parameters.WORKING_DIR, "snmptrapd_stdout")
+        self.snmptrapd_stderr_path = Path(tc_parameters.WORKING_DIR, "snmptrapd_stderr")
+
+    def start(self):
+        self.snmptrapd_proc = ProcessExecutor().start(
+            [
+                "snmptrapd", "-f",
+                "--disableAuthorization=yes",
+                self.port,
+                "-Lf", os.path.relpath(self.snmptrapd_log),
```

Comment:
`self.snmptrapd_log` object type is PosixPath, it does not have `startswith()` method (because of python2 compatibility). It should be converted to str.
`os.path.relpath(str(self.snmptrapd_log))` will work

Reply:
done

Reply:
thanks